feat: update response in users-following,followers per request

rubylo718 opened the pull request:
1. add introduction of the following/follower user
2. add followId to indicate th following/follower
issue #33

rubylo718 committed:
feat: update response in users-following,followers per request
1. add introduction of the following/follower user
2. add followId to indicate th following/follower
issue #33

ritachien reviewed (commented):
## Typos (sorry I miss this one last time.)  
* line 259/302:  next --> nest  

## `isFollowing`  refactor   
The following list of current user has already stored in `req.user`. No need to send another query to database. 
```js
// get `currentUserFollowingIds`
// replaced 264~269 & 307~312 by following codes
const currentUserFollowingIds = helpers.getUser(req).Followings.map(f => f.id)

// check if following
// replaced 272
isFollowing: currentUserFollowingIds.includes(follower.followingId)
// replaced 315
isFollowing: currentUserFollowingIds.includes(follower.followerId)
```

rubylo718 committed:
refactor: isFollowing in getUserFollowings and getUser followers
1. retreive current user following list from req.user instead of querying the db in composing isFollwing value
2. typo fixing

rubylo718 commented:
done!

```javascript
// check if following
// replaced 272
isFollowing: currentUserFollowingIds.includes(follower.followingId) // should be .includes(following.followingId)
// replaced 315
isFollowing: currentUserFollowingIds.includes(follower.followerId)
```

ritachien reviewed (approved):
Looks nice~
